File: ExamSystem_BE/JWT/DTOs/Auth/GoogleLoginRequest.cs
namespace JWT.DTOs.Auth
{
    /// <summary>
    /// Request body cho endpoint POST /api/auth/google-login.
    /// 
    /// Hỗ trợ 2 luồng xác thực linh hoạt:
    ///   1. IdToken (native/mobile) → verify bằng GoogleJsonWebSignature.
    ///   2. AccessToken (web) → gọi Google UserInfo API lấy email.
    /// Ít nhất 1 trong 2 phải có giá trị.
    /// </summary>
    public class GoogleLoginRequest
    {
        /// <summary>
        /// Google ID Token (JWT) — từ Google Sign-In SDK (native/mobile).
        /// Nullable trên Flutter Web vì plugin không trả về idToken.
        /// </summary>
        public string? IdToken { get; set; }

        /// <summary>
        /// Google Access Token — từ Google Sign-In SDK (web).
        /// Dùng khi IdToken null, BE gọi Google UserInfo API để lấy email.
        /// </summary>
        public string? AccessToken { get; set; }
    }
}

File: examsystem_fe/lib/features/auth/models/google_login_request.dart
/// ════════════════════════════════════════════════════════════════════════════
/// GoogleLoginRequest — Dữ liệu gửi lên BE để đăng nhập bằng Google.
///
/// Flow:
///   1. FE dùng Google Sign-In SDK lấy token từ Google.
///   2. FE gửi idToken + accessToken lên BE: POST /api/auth/google-login
///   3. BE ưu tiên verify idToken, fallback dùng accessToken gọi Google UserInfo API.
///   4. BE tìm user → trả AuthResponse (JWT).
///
/// Trên Flutter Web, idToken luôn null → BE dùng accessToken.
/// Trên native/mobile, idToken có giá trị → BE verify trực tiếp.
///
/// Đồng bộ với BE endpoint:
///   POST /api/auth/google-login
///   Body: { "idToken": "eyJ...", "accessToken": "ya29..." }
///   Response: AuthResponse (userId, fullName, email, accessToken, refreshToken)
/// ════════════════════════════════════════════════════════════════════════════
class GoogleLoginRequest {
  /// Google ID Token — JWT được Google cấp (null trên Web).
  final String idToken;

  /// Google Access Token — dùng khi idToken không có (Web fallback).
  final String? accessToken;

  const GoogleLoginRequest({
    required this.idToken,
    this.accessToken,
  });

  /// Chuyển thành JSON gửi lên BE.
  Map<String, dynamic> toJson() => {
    'idToken': idToken,
    if (accessToken != null && accessToken!.isNotEmpty)
      'accessToken': accessToken,
  };
}

